apispec/insolvency-delta-spec.yml

```diff
@@ -0,0 +1,160 @@
+post:
+  summary: Accepts an incoming insolvency delta, transforms it into an avro schema and puts it onto a Kafka topic.
+  requestBody:
+    content:
+      application/json:
+        schema:
+          $ref: '#/components/schemas/Insolvency'
+    required: true
+  responses:
+    "200":
+      description: Successfully produced message onto Kafka topic.
+    "400":
+      description: Bad request body - validation errors.
+    "401":
+      description: Unauthorised - missing api key in header.
+    "500":
+      description: Internal server error has occured.
+components:
+  schemas:
+    Insolvency:
+      required:
+      - company_number
+      - case_numbers
+      type: object
+      properties:
+        title:
+          type: string
+          enum: ['insolvency']
+        company_number:
+          maxLength: 10
+          type: string
+        case_numbers:
+          type: array
+          items:
+            $ref: '#/components/schemas/CaseNumber'
+
+    CaseNumber:
+      required:
+      - case_number
+      - case_type_id
+      - case_type
+      type: object
+      properties:
+        case_number:
+          type: integer
+        case_type_id:
+          type: integer
+          enum:
+          - 1
+          - 2
+          - 3
+          - 5
+          - 6
+          - 7
+          - 8
+          - 13
+          - 14
+          - 15
+          - 17
+        case_type:
+          type: string
+          enum: ['Members Voluntary Liquidation',
+                'Creditors Voluntary Liquidation',
+                'Compulsory Liquidation',
+                'Receiver/Manager',
+                'Administrative Receiver',
+                'Administration',
+                'Corporate Voluntary Arrangement',
+                'In Administration',
+                'CVA Moratorium',
+                'Foreign Insolvency',
+                'Moratorium']
+        mortgage_id:
+          type: integer
+        appointments:
+          type: array
+          items:
+            $ref: '#/components/schemas/Appointment'
+        wind_up_date:
+          type: string
+        dissolved_date:
+          type: string
+        dissolved_due_date:
+          type: string
+        sworn_date:
+          type: string
+        petition_date:
+          type: string
+        wind_up_conclusion_date:
+          type: string 
+        instrument_date:
+          type: string
+        admin_order_date:
+          type: string
+        discharge_admin_order_date:
+          type: string
+        report_date:
+          type: string
+        completion_date:
+          type: string
+        admin_start_date:
+          type: string
+        admin_end_date:
+          type: string
+        appointment_date:
+          type: string
+        end_date:
```

Comment:
I can't see how Chips would send 'end_date'. I could well be wrong but its not covered by the stored procedure which creates the data and its not in the sample data.

Reply:
I don't see it in the test data either but we have it under the mapping for the following case types:
- Moratorium
- CVA Moratoria

They're all optional so are we okay with leaving them in since they're specified?

Reply:
I'd say leave it for now but raise a question with the analysts about why it was included?

Reply:
Raised and will follow up